src/parser/Statement.ts

```diff
@@ -377,8 +377,16 @@ export class FunctionStatement extends Statement implements TypedefProvider {
         return this.func.transpile(state, nameToken);
     }
 
-    getTypedef(state: TranspileState) {
-        return this.func.getTypedef(state, this.name);
+    getTypedef(state: TranspileState, isClassMethod = false) {
```

Comment:
This doesn't belong here. This breaks the point of inheritance. ClassMethod should handle its own transpiling of annotations (or not), and same with FunctionStatement (because it wraps a FunctionExpression).